### PR TITLE
Numpy: Improves Redirects

### LINKED ISSUE #751: Numpy: Improve Redirects
Numpy queries make up a huge part of the Python Search Space so we need to get it right. It seems that there is a redirect issue in [output.txt](https://github.com/duckduckgo/zeroclickinfo-fathead/blob/e44b6520fa67ed6dca5f96dd9baf629205965c41/lib/fathead/numpy/output.txt). 

```
numpy.absolute	A										<section class="prog__container"><p>Calculate the absolute value element-wise.</p><pre><code>numpy.absolute(x[, out]) = <ufunc 'absolute'></code></pre><span class="prog__sub">Parameters:</span><pre><code>x : array_like\n    Input array.</code></pre><span class="prog__sub">Returns:</span><pre><code>absolute : ndarray\n    An ndarray containing the absolute value of each element in x.  For complex input, a + ib, the absolute value is .</code></pre></section>	https://docs.scipy.org/doc/numpy/reference/generated/numpy.absolute.html
numpy absolute	R	numpy.absolute		
```

It turns out `numpy absolute	R	numpy.absolute` is redundant, because the query parser ignores `.`. So this redirect would trigger on `numpy numpy absolute`. This needs to be fixed. If we could update the redirects to omit `numpy` that would be great. 

So the redirects should look as follows:

```
absolute	R	numpy.absolute
```

## Get Started
- [x] 1) Claim this issue by commenting below
- [x] 2) Review our [Contributing Guide](https://github.com/duckduckgo/zeroclickinfo-fathead/blob/master/CONTRIBUTING.md)
- [x] 3) [Set up your development environment](https://docs.duckduckhack.com/welcome/setup-dev-environment.html), and fork this repository
- [x] 4) Create a Pull Request

## Resources
- Join [DuckDuckHack Slack](https://quackslack.herokuapp.com/) to ask questions
- Join the [DuckDuckHack Forum](https://forum.duckduckhack.com/) to discuss project planning and Instant Answer metrics
- Read the [DuckDuckHack Documentation](https://docs.duckduckhack.com/) for technical help

---

Instant Answer Page: https://duck.co/ia/view/numpy
Maintainer: @ashishbhate

### PR DESCRIPTION
**Pull request by adityatandon007**
If this is a bug fix:
Fixes #751 

## Description of new Instant Answer, or changes

Fixes #751 : This PR improves the redirects by updating parse.py and output.txt file.

## Related Issues and Discussions

Fixes #751

## People to notify
@ashishbhate @pjhampton 

## Testing & Review
To be completed by Language Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [x] Title follows correct format (Specifies Instant Answer + Purpose)
- [x] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [x] Instant Answer page is correctly filled out and contains:
    - One topic for the Search Space Language (Java, Python, Scala, Ruby, etc.)
    - One topic from: Reference, Help, Libraries, Tools
        - Documentation Fatheads are considered "Reference"
    - Description, Data source, and 2+ example queries
    - Perl Module (e.g. "DDG::Fathead::PerlDoc" -- we only need a name, not an actual file)
    - Source Name (for "More at <source_name>" link)
    - Source Domain (must contain http:// or https:// -- can be the same as Data Source)
    - Source Info (used as Subtitle for each Article -- usually matches the IA Name)
    - 'Skip Abstract' is checked off
    - Source ID (ping @moollaza to assign one, once Fathead is ready for Beta deploy)

**Code**
- [x] Uniformly indented, well commented
- [x] Fetch.sh and Parse.sh run without errors
- [x] Output contains no blank lines, or multi-line entries
- [x] Fathead Tests are passing (run `$ duckpan test <fathead_id>`)
    - Tester should report any failures

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/numpy

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by adityatandon007** [comment]
@pjhampton One more thing the file size is too large. Can we do something more

File: lib/fathead/numpy/parse.py [hunk 1]
```diff
@@ -351,13 +351,12 @@ def get_redirects(title):
     redirects = set()
     title_split = title.split(".")
     if title_split[0] == PACKAGE_NAME:
-        redirects.add(PACKAGE_NAME + " " + ".".join(title_split[1:]))
-    else:
-        redirects.add(PACKAGE_NAME + "." + title)
+        title_split = title_split[1:]
+
+    redirects.add(".".join(title_split))
```
Comment (pjhampton) [inline lib/fathead/numpy/parse.py]: remember the period is dropped by the query. We don't need both lines, just one will do! 😄

Comment (adityatandon007) [inline lib/fathead/numpy/parse.py]: yeah I forgot that

Comment (adityatandon007) [inline lib/fathead/numpy/parse.py]: I'll correct and then make a commit

**Comment by adityatandon007** [comment]
@pjhampton made the changes

**Comment by pjhampton** [comment]
Thanks @adityatandon007! 💙  💛  💜 
Please remove the output.txt file and I will merge this

**Comment by ashishbhate** [comment]
Thanks @adityatandon007!

**Comment by pjhampton** [comment]
We will check back in about 2 weeks after this patch goes live and see what more we can do: https://forum.duckduckhack.com/t/numpy-fathead/1371

**Pull request merged**